packages/angular_devkit/build_angular/src/utils/process-bundle.ts

```diff
@@ -155,59 +155,21 @@ export async function process(options: ProcessBundleOptions): Promise<ProcessBun
     }
   }
 
-  if (options.optimize) {
-    if (downlevelCode) {
-      const minifyResult = terserMangle(downlevelCode, {
-        filename: downlevelFilename,
-        map: downlevelMap,
-        compress: true,
-      });
-      downlevelCode = minifyResult.code;
-      downlevelMap = minifyResult.map;
-    }
-
-    if (!options.ignoreOriginal) {
-      result.original = await mangleOriginal(options);
-    }
-  }
-
   if (downlevelCode) {
-    const downlevelPath = path.join(basePath, downlevelFilename);
-
-    let mapContent;
-    if (downlevelMap) {
-      if (!options.hiddenSourceMaps) {
-        downlevelCode += `\n//# sourceMappingURL=${downlevelFilename}.map`;
-      }
-
-      mapContent = JSON.stringify(downlevelMap);
-      await cachePut(mapContent, options.cacheKeys[CacheKey.DownlevelMap]);
-      fs.writeFileSync(downlevelPath + '.map', mapContent);
-    }
-
-    result.downlevel = createFileEntry(
-      path.join(basePath, downlevelFilename),
-      downlevelCode,
-      mapContent,
-      options.integrityAlgorithm,
-    );
-
-    await cachePut(
-      downlevelCode,
-      options.cacheKeys[CacheKey.DownlevelCode],
-      result.downlevel.integrity,
-    );
-    fs.writeFileSync(downlevelPath, downlevelCode);
+    result.downlevel = await processBundle({
+      ...options,
+      code: downlevelCode,
+      map: downlevelMap && JSON.stringify(downlevelMap),
```

Comment:
Can `map` be a string or a `RawSourceMap` here with a check in the `processBundle` function to handle either?  Sourcemaps can easily be in the megabyte range.  Serializing and immediately deserializing can have performance and memory usage concerns.

Reply:
Yeah make sense! Done.